cmd/kubetoken/validator.go

```diff
@@ -0,0 +1,232 @@
+	return &preReqCfg{
+		k8sClientProvider:    okteto.NewK8sClientProvider(),
+		oktetoClientProvider: okteto.NewOktetoClientProvider(),
+	}
+}
+
+// preReqValidator validates that all the pre-reqs to execute the command are met
+type preReqValidator struct {
+	ctxName string
+	ns      string
+
+	k8sClientProvider    k8sClientProvider
+	oktetoClientProvider oktetoClientProvider
+}
+
+// newPreReqValidator returns a new preReqValidator
+func newPreReqValidator(opts ...option) *preReqValidator {
+	cfg := defaultPreReqCfg()
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	return &preReqValidator{
+		ctxName:              cfg.ctxName,
+		ns:                   cfg.ns,
+		k8sClientProvider:    cfg.k8sClientProvider,
+		oktetoClientProvider: cfg.oktetoClientProvider,
+	}
+}
+
+// Validate validates that all the pre-reqs to execute the command are met
+func (v *preReqValidator) Validate(ctx context.Context) error {
+	oktetoLog.Info("validating pre-reqs for kubetoken")
+
+	ctx, cancel := context.WithTimeout(ctx, validationTimeout)
+	defer cancel()
+
+	err := newCtxValidator(v.ctxName, v.k8sClientProvider).Validate(ctx)
+	if err != nil {
+		return fmt.Errorf("invalid context: %w", err)
+	}
+
+	err = newOktetoSupportValidator(ctx, v.ctxName, v.ns, v.k8sClientProvider, v.oktetoClientProvider).Validate(ctx)
+	if err != nil {
+		return fmt.Errorf("invalid okteto support: %w", err)
+	}
+	return nil
+}
+
+// ctxValidator checks that the ctx use to execute the command is an okteto context
+// that has already being added to your okteto context
+type ctxValidator struct {
+	ctxName           string
+	k8sClientProvider k8sClientProvider
+	getContextStore   func() *okteto.OktetoContextStore
+	k8sCtxToOktetoURL func(ctx context.Context, k8sContext string, k8sNamespace string, clientProvider okteto.K8sClientProvider) string
+}
+
+func newCtxValidator(ctxName string, k8sClientProvider k8sClientProvider) *ctxValidator {
+	return &ctxValidator{
+		ctxName:           ctxName,
+		k8sClientProvider: k8sClientProvider,
+		getContextStore:   okteto.ContextStore,
+		k8sCtxToOktetoURL: okteto.K8sContextToOktetoUrl,
+	}
+}
+
+type errOktetoContextNotFound struct {
+	ctxName string
+}
+
+func (e errOktetoContextNotFound) Error() string {
+	return fmt.Sprintf("context '%s' not found in the okteto context store", e.ctxName)
+}
+
+type errIsNotOktetoCtx struct {
+	ctxName string
+}
+
+func (e errIsNotOktetoCtx) Error() string {
+	return fmt.Sprintf("context '%s' is not an okteto context", e.ctxName)
+}
+
+func (v *ctxValidator) Validate(ctx context.Context) error {
+	oktetoLog.Info("validating the context")
+	result := make(chan error, 1)
+	go func() {
+		if v.ctxName == "" {
+			result <- errEmptyContext
+			return
+		}
+		if !isURL(v.ctxName) {
+			v.ctxName = v.k8sCtxToOktetoURL(ctx, v.ctxName, "", v.k8sClientProvider)
+		}
+		okCtx, exists := v.getContextStore().Contexts[v.ctxName]
+		if !exists {
+			result <- errOktetoContextNotFound{v.ctxName}
```

Comment:
what do you think about making this log more explicit? something like:

```suggestion
	oktetoLog.Info("validating context for dynamic kubernetes token request")
```

Reply:
this comment is in regards to: `oktetoLog.Info("validating the context")`